File: ApiService/Infrastructure/Entities/SubmissionEntity.cs
namespace Infrastructure.Entities;

public class SubmissionEntity
{
    public Guid Id { get; set; }

    public Guid LabId { get; set; }

    public Guid StudentId { get; set; }

    public string Status { get; set; } = null!;

    public string SourceFileName { get; set; } = null!;

    public string MimeType { get; set; } = null!;

    public string StorageKey { get; set; } = null!;

    public DateTimeOffset SubmittedAt { get; set; }
}

File: ingestWorker/internal/storage/entity/submissionEntity.go
package entity

import "time"

type SubmissionEntity struct {
	tableName struct{} `pg:"submissions"`

	ID             string    `pg:"submission_id,pk"`
	LabID          string    `pg:"lab_id,notnull"`
	StudentID      string    `pg:"student_id,notnull"`
	Status         string    `pg:"status,notnull"`
	SourceFileName string    `pg:"source_file_name,notnull"`
	MimeType       string    `pg:"mime_type,notnull"`
	StorageKey     string    `pg:"storage_key,notnull"`
	SubmittedAt    time.Time `pg:"submitted_at,notnull"`
	CreatedAt      time.Time `pg:"created_at,notnull"`
}
